File: 9-palindrome-number/palindrome-number.swift
class Solution {

    func numberToArrayOfDigits(_ x: Int) -> [Int] {

        var result = [Int]()
        var value = x

        repeat {

            let digit = value % 10
            result.append(digit)
            
            value = value / 10

        } while value != 0

        return result
    }

    func isPalindrome(_ x: Int) -> Bool {
        
        if x < 0 {
            return false
        }
        
        let digits = numberToArrayOfDigits(x)
        let numberOfItems = digits.count

        for index in 0..<numberOfItems {

            let reversedIndex = numberOfItems - index - 1

            if digits[index] != digits[reversedIndex] {
                return false
            }
        }

        return true
    }
}

File: 9-palindrome-number/palindrome-number.py
class Solution:
    def reverse_number(x: int) -> int:
        temp = abs(x)
        result = 0

        while temp != 0:

            last_digit = temp % 10
            result = result * 10 + last_digit
            temp = temp // 10

        return result
    
    def isPalindrome(self, x: int) -> bool:
        return x >= 0 and x == Solution.reverse_number(x)